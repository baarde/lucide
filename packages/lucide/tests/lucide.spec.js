import { describe, it, expect } from 'vitest';
import { createIcons, icons } from '../src/lucide';
import path from 'path';

const ICONS_DIR = path.resolve(__dirname, '../../../icons');

describe('createIcons', () => {
  it('should read elements from DOM and replace it with icons', () => {
    document.body.innerHTML = `<i data-lucide="volume-2"></i>`;

    createIcons({ icons });

    expect(document.body.innerHTML).toMatchSnapshot();
  });

  it('should customize the name attribute', () => {
    document.body.innerHTML = `<i data-custom-name="volume-2"></i>`;

    createIcons({
      icons,
      nameAttr: 'data-custom-name',
    });

    const hasSvg = !!document.querySelector('svg');

    expect(hasSvg).toBeTruthy();
  });

  it('should add custom attributes', () => {
    document.body.innerHTML = `<i data-lucide="volume-2" class="lucide"></i>`;

    const attrs = {
      class: 'lucide lucide-volume-2 icon custom-class',
      fill: 'black',
    };

    createIcons({ icons, attrs });

    const element = document.querySelector('svg');
    const attributes = element.getAttributeNames();

    const attributesAndValues = attributes.reduce((acc, item) => {
      acc[item] = element.getAttribute(item);

      return acc;
    }, {});

    expect(document.body.innerHTML).toMatchSnapshot();

    expect(attributesAndValues).toEqual(expect.objectContaining(attrs));
  });

  it('should inherit elements attributes', () => {
    document.body.innerHTML = `<i data-lucide="sun" data-theme-switcher="light"></i>`;

    const attrs = {
      'data-theme-switcher': 'light',
    };

    createIcons({ icons });

    const element = document.querySelector('svg');
    const attributes = element.getAttributeNames();

    const attributesAndValues = attributes.reduce((acc, item) => {
      acc[item] = element.getAttribute(item);

      return acc;
    }, {});

    expect(attributesAndValues).toEqual(expect.objectContaining(attrs));
  });

  it('should read elements from DOM and replace icon with alias name', () => {
    document.body.innerHTML = `<i data-lucide="grid"></i>`;

    createIcons({ icons });

    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
