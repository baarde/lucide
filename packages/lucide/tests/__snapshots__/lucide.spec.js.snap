// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createIcons > should add custom attributes 1`] = `
<svg xmlns="http://www.w3.org/2000/svg"
     width="24"
     height="24"
     viewbox="0 0 24 24"
     fill="black"
     stroke="currentColor"
     stroke-width="2"
     stroke-linecap="round"
     stroke-linejoin="round"
     data-lucide="volume-2"
     class="lucide lucide-volume-2 icon custom-class"
>
  <path d="M11 5 6 9H2v6h4l5 4V5zM15.54 8.46a5 5 0 0 1 0 7.07M19.07 4.93a10 10 0 0 1 0 14.14">
  </path>
</svg>
`;

exports[`createIcons > should read elements from DOM and replace icon with alias name 1`] = `
<svg xmlns="http://www.w3.org/2000/svg"
     width="24"
     height="24"
     viewbox="0 0 24 24"
     fill="none"
     stroke="currentColor"
     stroke-width="2"
     stroke-linecap="round"
     stroke-linejoin="round"
     data-lucide="grid"
     class="lucide lucide-grid"
>
  <path d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2zM3 9h18M3 15h18M9 3v18M15 3v18">
  </path>
</svg>
`;

exports[`createIcons > should read elements from DOM and replace it with icons 1`] = `
<svg xmlns="http://www.w3.org/2000/svg"
     width="24"
     height="24"
     viewbox="0 0 24 24"
     fill="none"
     stroke="currentColor"
     stroke-width="2"
     stroke-linecap="round"
     stroke-linejoin="round"
     data-lucide="volume-2"
     class="lucide lucide-volume-2"
>
  <path d="M11 5 6 9H2v6h4l5 4V5zM15.54 8.46a5 5 0 0 1 0 7.07M19.07 4.93a10 10 0 0 1 0 14.14">
  </path>
</svg>
`;
